rancher2/import_rancher2_cloud_credentials.go

```diff
@@ -0,0 +1,39 @@
+package rancher2
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	managementClient "github.com/rancher/rancher/pkg/client/generated/management/v3"
+)
+
+func resourceRancher2CloudCredentialsImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	client, err := meta.(*Config).ManagementClient()
+	if err != nil {
+		return []*schema.ResourceData{}, err
+	}
+
+	cloudCredential := &CloudCredential{}
+	err = client.APIBaseClient.ByID(managementClient.CloudCredentialType, d.Id(), cloudCredential)
+	if err != nil {
+		return []*schema.ResourceData{}, err
+	}
+
+	drivers := []string{
+		amazonec2ConfigDriver,
+		azureConfigDriver,
+		digitaloceanConfigDriver,
+		googleConfigDriver,
+		s3ConfigDriver,
+		vmwarevsphereConfigDriver,
+	}
+
+	// Missing "driver" field in api.
+	for _, driver := range drivers {
+		d.Set("driver", driver)
+		err = flattenCloudCredential(d, cloudCredential)
+		if err != nil {
+			return []*schema.ResourceData{}, err
+		}
+	}
```

Comment:
Hi @rawmind0,

> Cloud credential sensitive data can't be read from Rancher API.

This doesn't matter, this is something you have to fix after the import anyway.
The main goal , for me at least, is to get the main structure imported.

> How is this checking that the used driver is the proper one??

For other resources i can see there is a `driver: ...` field in the api of rancher.
Cloud Credential does not have this, this is why i just use a loop with all supported resources.
The better solution would be to add `driver` to the rancher api this way the code can target only one resource with `get('driver')`

That said, the empty resources return nothing and only the resource configured will be added.
Must be a hidden feature ;-)

> On the import command the driver should be specified with the id and split within the resourceRancher2CloudCredentialsImport function

I haven't seen any use of this before, can you point me to some examples?

Reply:
> This doesn't matter, this is something you have to fix after the import anyway.
The main goal , for me at least, is to get the id imported. Because i can't change the statefile.

What do you mean with `...this is something you have to fix after the import anyway...` ?? The cloud credential should be added to the tf file before import and once imported, shouldn't generate diffs on apply.

> For other resources i can see there is a driver: ... field in the api of rancher.
Cloud Credential does not have this, this is why i just use a loop with all supported resources.
The better solution would be to add driver to the rancher api this way the code can target only one resource with get('driver')
> 
> That said, the empty resources return nothing and only the resource configured will be added.
Must be a hidden feature ;-)

May be i'm loosing something, but i don't get you point on this. The code is assigning the driver value to the resource data in every iteration, https://github.com/rancher/terraform-provider-rancher2/pull/847/files#diff-0308de0f04c805a9ddb5d67f5c1fe0bc5092e1af4ff918d0fd60136c8f4f412aR31 What is the final value of `driver` resourceData?? 

> I haven't seen any use of this before, can you point me to some examples?

Sure, you can take a look to [import_rancher2_app_v2](https://github.com/rancher/terraform-provider-rancher2/blob/master/rancher2/import_rancher2_app_v2.go#L7) . The import id is composed by `<CLUSTER_ID>.<APP_V2_NAME>` https://registry.terraform.io/providers/rancher/rancher2/latest/docs/resources/app_v2#import . Here can be done something similar, `<CLOUD_CREDENTIAL_ID>.<DRIVER>` and forget the loop.

Reply:
> What do you mean with ...this is something you have to fix after the import anyway... ?? The cloud credential should be added to the tf file before import and once imported, shouldn't generate diffs on apply.

https://spacelift.io/blog/importing-exisiting-infrastructure-into-terraform
We wanted to migrate from our own ansible modules to terraform.

*  We have an existing cluster, with filled resources.
* We want to create a terraform state that has no changes.
* I want to import the existing id into the terraform state.
* And then work away all the changes.
* I have the credentials, i don't need them from rancher.

Reply:
> What is the final value of driver resourceData??

There is no final value for the driver. That will be determent with the next `terraform plan/apply`.
This is just importing anything that will be found. And rancher does only allow one resource.
This will, for now, will alwasy go correctly :D

Reply:
> `<CLOUD_CREDENTIAL_ID>.<DRIVER>` and forget the loop.

Good suggestion, will change the pull tomorrow.